pkg/plugins/policies/meshtrafficpermission/api/v1alpha1/validator.go

```diff
@@ -0,0 +1,50 @@
+package v1alpha1
+
+import (
+	common_proto "github.com/kumahq/kuma/api/common/v1alpha1"
+	"github.com/kumahq/kuma/pkg/core/validators"
+	matcher_validators "github.com/kumahq/kuma/pkg/plugins/policies/matchers/validators"
+)
+
+func (r *MeshTrafficPermissionResource) validate() error {
+	verr := validators.ValidationError{}
+	path := validators.RootedAt("spec")
+
+	if targetRef := r.Spec.GetTargetRef(); targetRef != nil {
+		targetRefErr := matcher_validators.ValidateTargetRef(path.Field("targetRef"), targetRef, &matcher_validators.ValidateTargetRefOpts{
+			SupportedKinds: []common_proto.TargetRef_Kind{
+				common_proto.TargetRef_Mesh,
+				common_proto.TargetRef_MeshSubset,
+				common_proto.TargetRef_MeshService,
+				common_proto.TargetRef_MeshServiceSubset,
+				common_proto.TargetRef_MeshGatewayRoute,
+				common_proto.TargetRef_MeshHTTPRoute,
+			},
+		})
+		verr.AddError("", targetRefErr)
+	}
```

Comment:
This at least can be partially pregenerated in the bootstrap.

Reply:
Same, I'll update this in the follow-up PR